components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java

```diff
@@ -4724,6 +4735,46 @@ public API getAPIbyUUID(String uuid, String organization) throws APIManagementEx
         }
     }
 
+    @Override
+    public Map<String, Object> searchPaginatedAPIsByFQDN(String searchQuery, String tenantDomain, int start, int end) throws APIManagementException {
+        Map<String, Object> result = new HashMap<String, Object>();
+
+        Organization org = new Organization(organization);
+        String[] roles = APIUtil.getFilteredUserRoles("admin");
```

Comment:
I think we can get the roles before hand and keep and use it without retrieving them every time but check the fact where these roles can be dynamically changed or not

Reply:
Implemented differently in the new implementation of the method